### PR TITLE
feat: do not reload list of proposals on back

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

To make the transition between "Proposal detail" and "Proposals list - Voting" pages smooth in case of "back" action, the proposals shall not be fetched again if already exist.

Scenario example: 

- Neurons > Voting > Fetch proposals > select Proposal > Detail > Back > No fetch - display existing proposals from store
- Proposal details > Fetch proposal > Back > Voting > Fetch proposals
- Neurons > Voting -> Fetch proposals  > Neurons > Voting > Fetch proposals

# Changes

- add `referrerPath` to route store
- do not reset proposals store on mount in `Proposals.svelte` if referrer is "Proposal detail" page and store not empty
- fix `navigate` and `replace` called at the same time on back in `ProposalDetail.svelte`

### CODE REVIEW

File: frontend/svelte/src/lib/stores/route.store.ts [hunk 1]
```diff
@@ -13,6 +14,12 @@ export interface RouteStore {
  * - update: update the state value
  * - navigate: update the state value and push the new route to the browser history i.e. update browser url with an entry in the navigation stack
  * - replace: update the state value and replace the route in the browser history i.e. update browser url without modifying the navigation stack
+ *
+ * The store contains following information:
+ * - path: the application path - i.e. the page (see AppPath) that is displayed
+ * - referrerPath: the page - i.e. the path - that linked to the current path. For example, if navigating from "Accounts" to "Voting" page, the referrerPath is `/#/accounts`. If the referrer is not the app, then is value is undefined.
+ * - isKnownPath: tells if the path is a path that is supported by the app - e.g. useful to display a 404
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/stores/route.store.ts:21]: Just as reminder: to mock the flutter dapp we navigate to accounts/ on unknown paths.

File: frontend/svelte/src/lib/stores/route.store.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { pushHistory, replaceHistory, routePath } from "../utils/route.utils";
 
 export interface RouteStore {
   path: AppPath | string;
+  referrerPath?: AppPath | string;
```
Comment (mstrasinskis) [inline frontend/svelte/src/lib/stores/route.store.ts:8]: Does it makes sense to set the default value for referredPath to `document.referrer`?

File: frontend/svelte/src/routes/ProposalDetail.svelte [hunk 3]
```diff
@@ -44,23 +38,28 @@
...
-
       console.error(error);
       toastsStore.show({
         labelKey: "error.proposal_not_found",
         level: "error",
         detail: `id: "${proposalId}"`,
       });
 
-      // Wait a bit before redirection so the user recognizes on which page the error occures
+      // Wait a bit before redirection so the user recognizes on which page the error occurs
```
Comment (mstrasinskis) [inline frontend/svelte/src/routes/ProposalDetail.svelte:48]: 🤦‍♂️ thanks

**Review by mstrasinskis — COMMENTED** [review]
The code is definitely cleaner, but I'm still not sure about single `unsubscribe`. For example if I'm on the `http://127.0.0.1:8080/#/proposal/50` and manually update the url to be http://127.0.0.1:8080/#/proposal/`63` there is no update of the data or any other feedback to the user. And this is not the flutter dapp behaviour.

File: frontend/svelte/src/lib/stores/route.store.ts (hunk 2 above)
Comment (peterpeterparker) [reply frontend/svelte/src/lib/stores/route.store.ts:8]: I was my original idea but `document.referrer` does not returns the path

Navigate from `https://nns.ic0.app/#/proposals` to `https://nns.ic0.app/#/neurons` -> referrer = `https://nns.ic0.app/v2/`

Comment (peterpeterparker) [reply frontend/svelte/src/lib/stores/route.store.ts:8]: Navigate from `http://localhost:8080/#/proposal/76` to `http://localhost:8080/#/proposals` -> referrer = `''`

File: frontend/svelte/src/routes/ProposalDetail.svelte (hunk 3 above)
Comment (peterpeterparker) [reply frontend/svelte/src/routes/ProposalDetail.svelte:48]: haha I am the one that generally make such mistakes, happy to have spotted one 😉

**Comment by peterpeterparker** [comment]
> The code is definitely cleaner, but I'm still not sure about single unsubscribe. For example if I'm on the .../#/proposal/50 and manually update the url to be ...#/proposal/63 there is no update of the data or any other feedback to the user. And this is not the flutter dapp behaviour.

You are right. I revert the code but I add a `unsubscribe` to `goBack` and that should do the tricks for both use cases.

**Review by mstrasinskis — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**